cb/cb_genesis2.py

```diff
@@ -1,48 +1,20 @@
-import argparse
-from common.genesis_wrapper import define_genesis_generator
+from common.genesis_wrapper import GenesisWrapper
+from common.generator_interface import GeneratorInterface
 
 
-define_cb_wrapper = define_genesis_generator(
-    top_name="cb",
-    input_files=["genesis/cb.vp"],
-    width=16,
-    num_tracks=10,
-    feedthrough_outputs="1111101111",
-    has_constant=1,
-    default_value=0
-)
-
-
-def create_parser():
-    parser = argparse.ArgumentParser()
-    parser.add_argument("infile", type=str)
-    parser.add_argument("--width", type=int, default=32)
-    parser.add_argument("--num_tracks", type=int, default=10)
-    parser.add_argument("--feedthrough_outputs", type=str, default=("1"*10))
-    parser.add_argument("--has_constant", type=int, default=0)
-    parser.add_argument("--default_value", type=int, default=0)
-    return parser
-
-
-def main(args):
-    # Check some of the inputs.
-    for c in args.feedthrough_outputs:
-        assert c == "1" or c == "0"
-
-    cb = define_cb_wrapper(width=args.width,
-                           num_tracks=args.num_tracks,
-                           feedthrough_outputs=args.feedthrough_outputs,
-                           has_constant=args.has_constant,
-                           default_value=args.default_value,
-                           input_files=[args.infile])
-    print(cb)
+interface = GeneratorInterface()\
+            .register("width", int, 16)\
+            .register("num_tracks", int, 10)\
+            .register("feedthrough_outputs", str, "1"*10)\
+            .register("has_constant", int, 0)\
+            .register("default_value", int, 0)
```

Comment:
Nice, seems like a reasonable way to specify the parameters with types and default values.